From 15023e58f792d88c595c9c4e15688b3f12e104d1 Mon Sep 17 00:00:00 2001
From: xeno27 <20142175+xeno27@users.noreply.github.com>
Date: Mon, 23 Feb 2026 13:58:50 +0000
Subject: [PATCH] slcan_attach: Works only with sllin for now. (for actual
 cantutils)

---
 slcan_attach.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/slcan_attach.c b/slcan_attach.c
index e72610a..7a33deb 100644
--- a/slcan_attach.c
+++ b/slcan_attach.c
@@ -55,6 +55,8 @@
 #include <termios.h>
 #include <unistd.h>
 
+#define LDISC_N_SLLIN 28
+
 void print_usage(char *prg)
 {
 	fprintf(stderr, "%s - userspace tool for serial line CAN interface driver SLCAN.\n", prg);
@@ -92,7 +94,7 @@ void print_usage(char *prg)
 int main(int argc, char **argv)
 {
 	int fd;
-	int ldisc = N_SLCAN;
+	int ldisc = LDISC_N_SLLIN;
 	int detach = 0;
 	int waitkey = 0;
 	int send_open = 0;
-- 
2.47.3

